libs/community/langchain_community/llms/huggingface_endpoint.py

```diff
@@ -173,16 +174,20 @@ def validate_environment(cls, values: Dict) -> Dict:
                 "Could not import huggingface_hub python package. "
                 "Please install it with `pip install huggingface_hub`."
             )
-        try:
-            huggingfacehub_api_token = get_from_dict_or_env(
-                values, "huggingfacehub_api_token", "HUGGINGFACEHUB_API_TOKEN"
-            )
-            login(token=huggingfacehub_api_token)
-        except Exception as e:
-            raise ValueError(
-                "Could not authenticate with huggingface_hub. "
-                "Please check your API token."
-            ) from e
+        huggingfacehub_api_token = values["huggingfacehub_api_token"] or os.getenv(
+            "HUGGINGFACEHUB_API_TOKEN"
+        )
+        if huggingfacehub_api_token is not None:
+            try:
+                huggingfacehub_api_token = get_from_dict_or_env(
+                    values, "huggingfacehub_api_token", "HUGGINGFACEHUB_API_TOKEN"
+                )
```

Comment:
These lines are leftovers, I suppose. Also, I guess a try-catch is still needed to re-throw the exception that the client might raise if the token was required, but not provided.